src/compile/projection/assemble.ts

```diff
@@ -26,7 +24,7 @@ export function assembleProjectionForModel(model: Model): VgProjection[] {
   }
 
   const projection = component.combine();
-  const {name, ...rest} = projection;
+  const {name, ...rest} = projection;  // name is always defined
```

Comment:
> // name is always defined

This confuses me, but Dom just explained to me the real reason behind this. 

Please change to .. "we need to extract `name` so that it is always present in the output and pass TS type validation" or something similar.